app/routines/exercises/import/assessments.rb

```diff
@@ -25,79 +16,170 @@ def exec(filename:, book_uuid:)
+
+              initialized = true
             end
 
-            exercise = Exercise.new
+            row_number = row_index + 1
 
-            exercise.tags = [
-              "assessment:#{row[pre_or_post_index].downcase == 'pre' ? 'preparedness' : 'practice'
-                }:https://openstax.org/orn/book:page/#{book_uuid}:#{page_uuid}",
-              "context-cnxmod:#{page_uuid}"
-            ]
+            # Using row_index here because dealing with the previous row
+            if multi_step_index.nil? || multi_step.nil? || row[multi_step_index] != multi_step
+              save.call(exercise, row_index)
+
+              multi_step = row[multi_step_index] unless multi_step_index.nil?
+
+              exercise = Exercise.new
+
+              page_uuid = if uuid_index.nil? || row[uuid_index].blank?
+                page_uuid_by_book_location[row[section_index].split('.').map(&:to_i)] \
+                  unless section_index.nil? || row[section_index].blank?
+              else
+                row[uuid_index]
+              end
+              if page_uuid.blank?
+                Rails.logger.warn { "Row ##{row_number} has no associated page in the book" } \
+                  unless uuid_index.nil? && section_index.nil?
+              else
+                exercise.tags << "context-cnxmod:#{page_uuid}"
+                exercise.tags << "assessment:#{row[pre_or_post_index].downcase == 'pre' ? 'preparedness' : 'practice'
+                  }:https://openstax.org/orn/book:page/#{book_uuid}:#{page_uuid}" \
+                  unless pre_or_post_index.nil? || row[pre_or_post_index].blank?
+              end
+
+              unless teks_index.nil? || row[teks_index].blank?
+                teks = row[teks_index].split(/,|;/).map(&:strip)
+                teks.each { |tek| exercise.tags << "teks:#{tek}" }
+              end
+              unless machine_teks_index.nil? || row[machine_teks_index].blank?
+                machine_teks = row[machine_teks_index].split(/,|;/).map(&:strip)
+                machine_teks.each { |tek| exercise.tags << "machine-teks:#{tek}" }
+              end
+              exercise.tags << "raise-content-id:#{row[raise_id_index]}" \
+                unless raise_id_index.nil? || row[raise_id_index].blank?
+
+              exercise.publication.authors << Author.new(user: author)
+              exercise.publication.copyright_holders << CopyrightHolder.new(user: copyright_holder)
+
+              exercise.publication.publication_group.nickname = row[nickname_index] unless nickname_index.nil?
+
+              exercise.stimulus = parse(row[background_index], exercise) unless background_index.nil?
+            else
+              Rails.logger.info { "Imported row ##{row_index} - Multi-step exercise" }
+            end
 
             question = Question.new
+            question.sort_position = row[block_index].to_i + 1 unless block_index.nil?
```

Comment:
Before they'd be ordered by the order in the spreadsheet, right? So this is a new field to control the order?

Reply:
Yes, though it's still optional